Code/Source/Entity/Terrain/TerrainData.cpp

```diff
@@ -12,95 +12,19 @@
     {
-    }
-
-    void TerrainEntityManagerSystemComponent::Activate()
-    {
-        UpdateWorldBounds();
-        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusConnect();
-    }
-
-    void TerrainEntityManagerSystemComponent::Deactivate()
-    {
-        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusDisconnect();
-    }
-
-    void TerrainEntityManagerSystemComponent::Reflect(AZ::ReflectContext* context)
-    {
-        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
-        {
-            serializeContext->Class<TerrainEntityManagerSystemComponent, AZ::Component>()->Version(0);
-
-            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
-            {
-                editContext->Class<TerrainEntityManagerSystemComponent>("Terrain Entity Manager", "Manages the RGL Terrain entity.")
-                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
-                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("System"))
-                    ->Attribute(AZ::Edit::Attributes::Category, "RGL")
-                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true);
-            }
-        }
-    }
-
-    void TerrainEntityManagerSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
-    {
-        provided.push_back(AZ_CRC_CE("TerrainEntityManagerService"));
-    }
-
-    void TerrainEntityManagerSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
-    {
-        incompatible.push_back(AZ_CRC_CE("TerrainEntityManagerService"));
-    }
-
-    void TerrainEntityManagerSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
-    {
-        required.push_back(AZ_CRC_CE("RGLService"));
-    }
-
-    void TerrainEntityManagerSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
-    {
-    }
-
-    void TerrainEntityManagerSystemComponent::EnsureManagedEntityDestroyed()
-    {
-        if (m_rglEntity)
-        {
-            RGL_CHECK(rgl_entity_destroy(m_rglEntity));
-            m_rglEntity = nullptr;
-        }
-
-        if (m_rglMesh)
-        {
-            RGL_CHECK(rgl_mesh_destroy(m_rglMesh));
-            m_rglMesh = nullptr;
-        }
-    }
-
-    void TerrainEntityManagerSystemComponent::UpdateWorldBounds()
-    {
-        AZ::Aabb newWorldBounds = AZ::Aabb::CreateFromPoint(AZ::Vector3::CreateZero());
-        AzFramework::Terrain::TerrainDataRequestBus::BroadcastResult(
-            newWorldBounds, &AzFramework::Terrain::TerrainDataRequests::GetTerrainAabb);
-
         if (newWorldBounds == m_currentWorldBounds)
         {
             // This is only for world bound creation, not for update.
-            return;
+            return false;
         }
 
         m_currentWorldBounds = newWorldBounds;
```

Comment:
Not sure, but I think `m_currentWorldBounds` should be updated at the end of this function (when returning true). Case:
The first invocation of this function returns false because of [this condition](https://github.com/RobotecAI/o3de-rgl-gem/pull/30/files#diff-2b0ca9c29e059f6c103b55316e7a8b49785d15b1d44e0c46a5bcac0290bb4420R38), but `m_currentWorldBounds` is already updated. Next invocations will always return false because `newWorldBounds == m_currentWorldBounds`.

Reply:
Is it not like this function only updates bounds, and then vertex heights are updated because whole region is dirty? Or something like that?

Reply:
This is a good suggestion but I don't see the "will always return false" part. Irrelevant of this, if the bounds are updated, true must be returned as the Terrain Data was technically modified.